tests/secret-providers/gcp_secret_manager_workload_identity/gcp_secret_manager_workload_identity_test.go

```diff
@@ -0,0 +1,453 @@
+//go:build e2e
+// +build e2e
+
+package gcp_secret_manager_workload_identity_test
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"os"
+	"testing"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	"cloud.google.com/go/secretmanager/apiv1/secretmanagerpb"
+	"github.com/joho/godotenv"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/option"
+	"k8s.io/client-go/kubernetes"
+
+	. "github.com/kedacore/keda/v2/tests/helper"
+)
+
+// Load environment variables from .env file
+var _ = godotenv.Load("../../.env")
+
+const (
+	testName = "gcp-secret-manager-workload-identity-test"
+)
+
+var (
+	testNamespace              = fmt.Sprintf("%s-ns", testName)
+	deploymentName             = fmt.Sprintf("%s-deployment", testName)
+	scaledObjectName           = fmt.Sprintf("%s-so", testName)
+	triggerAuthenticationName  = fmt.Sprintf("%s-ta", testName)
+	secretName                 = fmt.Sprintf("%s-secret", testName)
+	postgreSQLStatefulSetName  = "postgresql"
+	postgresqlPodName          = fmt.Sprintf("%s-0", postgreSQLStatefulSetName)
+	postgreSQLUsername         = "test-user"
+	postgreSQLPassword         = "test-password"
+	postgreSQLDatabase         = "test_db"
+	postgreSQLConnectionString = fmt.Sprintf("postgresql://%s:%s@postgresql.%s.svc.cluster.local:5432/%s?sslmode=disable",
+		postgreSQLUsername, postgreSQLPassword, testNamespace, postgreSQLDatabase)
+	minReplicaCount = 0
+	maxReplicaCount = 2
+
+	gcpKey                = os.Getenv("TF_GCP_SA_CREDENTIALS")
+	creds                 = make(map[string]interface{})
+	errGcpKey             = json.Unmarshal([]byte(gcpKey), &creds)
+	projectID             = creds["project_id"]
+	secretManagerSecretID = "connectionString"
```

Comment:
I think that we should use something to make the ID unique. The other e2e tests uses the same and it can produce conflicts if both are executed together

Reply:
This ID represents the name given to the secret created in GCP Secret Manager. AFAIK no other scaler/test uses GCP Secret Manager, right?

Reply:
No, but there are 2 tests using it, with and without workload identity, each one in its own namespace

Reply:
Good observation, I have updated the secret ID as `connectionStringWorkloadIdentity` for the workload identity test.